zerver/lib/subscription_info.py

```diff
@@ -438,6 +438,9 @@ def has_metadata_access_to_previously_subscribed_stream(
     if stream_dict["invite_only"]:
         return user_profile.is_realm_admin
 
+    if stream_dict["archived"]:
+        return True
+
     return True
```

Comment:
Can you explain what this test does, given that you `return True` just after anyway?

Reply:
Yes you are right. Fixed that.